controller/ticket-reports/tagWiseTicketSummaryCtrl.js

```diff
@@ -0,0 +1,233 @@
+            });
+        };
+
+        $scope.moment = moment;
+
+
+        $scope.obj = {
+            startDay: moment().format("YYYY-MM-DD"),
+            endDay: moment().format("YYYY-MM-DD")
+        };
+
+        $scope.tagSummaryDetails = [];
+
+        $scope.tagList = [];
+
+        $scope.querySearch = function (query)
+        {
+            var emptyArr = [];
+            if (query === "*" || query === "") {
+                if ($scope.tagList) {
+                    return $scope.tagList;
+                }
+                else {
+                    return emptyArr;
+                }
+
+            }
+            else {
+                if ($scope.tagList)
+                {
+                    return $scope.tagList.filter(function (item)
+                    {
+                        var regEx = "^(" + query + ")";
+
+                        if (item.name)
+                        {
+                            return item.name.match(regEx);
+                        }
+                        else
+                        {
+                            return false;
+                        }
+
+                    });
+                }
+                else {
+                    return emptyArr;
+                }
+            }
+
+        };
+
+
+        var getTagList = function () {
+            $scope.tagList = [];
+            ticketReportsService.getTagList().then(function (tagList) {
+                if (tagList && tagList.Result)
+                {
+                    $scope.tagList.push.apply($scope.tagList, tagList.Result);
+
+                }
+
+                ticketReportsService.getCategoryList().then(function (categoryList)
+                {
+                    if (categoryList && categoryList.Result)
+                    {
+                        $scope.tagList.push.apply($scope.tagList, categoryList.Result);
+                    }
+
+
+                }).catch(function (err) {
+                    loginService.isCheckResponse(err);
+
+                });
+
+
+            }).catch(function (err) {
+                loginService.isCheckResponse(err);
+            });
+        };
+
+        getTagList();
+
+
+        $scope.getTicketTagSummary = function ()
+        {
+            $scope.tagSummaryDetails = [];
+            $scope.obj.isTableLoading = 0;
+            var momentTz = moment.parseZone(new Date()).format('Z');
+            momentTz = momentTz.replace("+", "%2B");
+
+            var startDate = $scope.obj.startDay + ' 00:00:00' + momentTz;
+            var tempEndDate = $scope.obj.endDay;
+
+            var endDate = moment(tempEndDate).add(1, 'days').format('YYYY-MM-DD') + ' 00:00:00' + momentTz;
+
+            try
+            {
+                ticketReportsService.getTicketSummaryTagWise(startDate, endDate).then(function (ticketSummaryResp)
+                {
+                    if (ticketSummaryResp && ticketSummaryResp.Result && ticketSummaryResp.Result.length > 0)
```

Comment:
check isSuccess field of the response for success print appropriate message if service has failed